### PR TITLE
Add test for custom compression encodings that we should not error on

### PR DESCRIPTION
**Pull request by MStoykov**

### CODE REVIEW

**Comment by MStoykov** [comment]
This again is an old change that I found and decided to publish and pull request :)

**Comment by MStoykov** [comment]
we already needed to merge master in new-schedulers either way :D

File: js/modules/k6/http/request_test.go [hunk 1]
```diff
@@ -354,6 +354,22 @@ func TestRequestAndBatch(t *testing.T) {
...
+				assert.NoError(t, err)
+			}))
+
+			_, err := common.RunString(rt, sr(`
+				let res = http.get("HTTPBIN_URL/customcompression");
+				if (res.json()["custom"] != true) {
+					throw new Error("unexpected body data: " + res.body)
+				}
+			`))
+			assert.NoError(t, err)
```
Comment (imiric) [inline js/modules/k6/http/request_test.go:371]: Hmm I wonder if we should be smarter about this and not just ignore the header if we don't support the encoding. Maybe the correct behavior would be to send `Accept-Encoding` and throw an error if the server returns something else?

Comment (MStoykov) [reply js/modules/k6/http/request_test.go:371]: The whole point of this is that as long as we don't actually touch the response, the user can use whatever custom encoding/decoding they like. We have no mechanism to give that any other way, so I think this is better ... although it is possible that I am wrong

**Pull request merged**